### PR TITLE
Master

### LINKED ISSUE #37: Update to use pytest symbols plugin
@pskergan 

Update code used to define command line options for bigip values to use symbols instead.

### PR DESCRIPTION
**Pull request by pskergan**
@dflanigan  @szakeri 
Fixes #37 

The goal was to eliminate the command line options for environmental data and to use symols data instead.  

Note that only lbaasv2 testing uses this repository, and the test_env.json file required for this change is that for test/functional in f5-openstack-lbaasv2-driver; however, any json file with the referenced keys can be used here with symbols by other projects.

### CODE REVIEW

**Comment by zancas** [comment]
I think @pjbreaux is also using this repo, so we need to sync with him regarding release branch policies etc.   I'd like him to review the last three commits (which correspond to a PR _into_ liberty, before we move forward).

File: f5_os_test/infrastructure.py [hunk 1]
```diff
@@ -14,23 +14,144 @@
...
+    '''Heat client manager fixture.'''
+    config_dict = {
+        'endpoint': symbols.heatclient_url,
+        'token':    get_auth_token
+    }
+    return heatclient_pollster(**config_dict)
+
+
+@pytest.fixture
+def keystoneclientmanager(symbols, keystoneclient_pollster):
```
Comment (zancas) [inline f5_os_test/infrastructure.py:139]: I suggest we make "symbols" be the last positional parameter in all cases where that's feasible (doesn't conflict with some overriding policy).   Waddya' think?

**Comment by zancas** [comment]
This PR looks great to me!  I'm excited to migrate to config files for test parametrization! 

I'd like to see as much standardization of function params as possible: 

 Since "symbols" is a positional parameter we should put it in the same position where convenient.    I think making it the last _positional_ parameter might be logical since it's a container for variable values.

  What do you think @pskergan ?

**Comment by pskergan** [comment]
That's a good idea.  The other option is to disregard symbols as a fixture, but to import the data as a global container:    from pytest  import  symbols  as  symbols_data,   and then use symbols_data wherever needed.  We currently use both options. We use symbols as a fixture where fixtures can be employed and "symbols_data"  everywhere else.  It might just be simpler and more convenient to use the one option that applies everywhere.  Thoughts?

**Comment by pjbreaux** [comment]
Looking now...

**Comment by zancas** [comment]
I strongly believe it _must_ be all one way, or all the other.   If we can _always_ run:

`from pytest import symbols as symbols_data`

then I'm +1 on always doing it in that fashion.    Have @mattgreene and/or @pjbreaux weighed in on this?

I am -1 on doing it 2 different ways unless we have to.

Again, I don't care so much about which method we choose, but I strongly believe that there should be only 1 way we do it.   This will significantly enhance readability.

**Comment by zancas** [comment]
OK, I think I've shared all my opinions so I'm going to drop the assignment-hot-potato into @pjbreaux 's lap because I want his seal of approval/opinions before this thing lands.

**Comment by pjbreaux** [comment]
Let me just run the functional tests with the new world order...

**Comment by zancas** [comment]
Oh...  also I think @dflanigan and @mattgreene need to agree-or-disagree before we can merge.   The decisions we make here will have major effects on our test infrastructure moving forward.

**Comment by pjbreaux** [comment]
I'll punt it once I'm through

**Comment by pjbreaux** [comment]
@phil:

In the setup.py, the heat_utils are there: 'heat_utils = f5_os_test.heat_client_utils'

But there is no heat_client_utils file in the f5_os_test/ directory.

**Comment by zancas** [comment]
Yeah @pjbreaux @phil the removal of that file is the specific issue that I thought might indicate some sort of problem.

**Comment by pjbreaux** [comment]
I believe we need to have a common subset of the test_env variables that exist for all tests in openstack. These would have standardized naming conventions such as 'os_username' and 'bigip_admin_password'. This would take the guess work out of trying to figure out if @zancas' 'admin_password' is for openstack or for the BigIP. This means we would have to aggregate test_env files (if we can't already).

I'm less concerned about how a repo chooses to access that symbols data and more concerned about all the naming conventions of the data itself. But if I had to vote, I think having the global import, as @pskergan suggested, is my preferred solution.

I'm done with my testing and I will sign off as long as the heat_client_utils gets into this PR.

Punting this to @mattgreene

**Comment by zancas** [comment]
In reference to @pjbreaux 's vote for 'global import style':

https://www.youtube.com/watch?v=urvlIV-y9DY&t=0m40s

**Comment by pjbreaux** [comment]
Merging!

**Pull request merged**